eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -17,12 +17,23 @@ steps:
     - pwsh: |
         git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
         $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
-
-        if (!(Test-Path $repoPath)) { mkdir $repoPath }
-        Remove-Item -v -r $repoPath
-        Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
-        Get-ChildItem -r $repoPath
-
+        if ($PatchFilesLocation)
+        {
+          pushd ${{ repo }}
+          $results = Get-ChildItem $PatchFilesLocation | % { git am $_.FullName }
```

Comment:
What ends up in the $results array? Wouldn't it be all the outputted lines?

Reply:
This git am works correctly it $result will contain a line showing information for each applied patch file which should equal the number of patch files. If the number of lines in result don't match the number of patch files then an error occurred somewhere.

Reply:
We should try to use exit codes if they are correctly set as those are a little better then depending on the console output (especially the number of lines in the output).

Reply:
There is still scenario where the $LASTEXITCODE is 0 but no patch file was created. I will make use of both.